examples/help-listeners

```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env node
+
+// const program = require('commander'); // (normal include)
+const program = require('../'); // include commander in git clone of commander repo
+
+program
+  .version('0.0.1')
+  .option('-b, --bar', 'enable some bar')
+  .on('help:header', () => {
+    console.log('Custom Header\n');
+  })
+  .on('help:body', (instance) => {
+    console.log('------------');
+    console.log(instance.helpInformation().replace(/\n$/, '\n------------\n'));
+  })
+  .on('help:footer', () => {
+    console.log(`Commander - ${new Date().getFullYear()}`);
+  })
```

Comment:
The replace and `\n` are a bit confusing here. I eventually recalled that `helpInformation()` has a trailing return for legacy reasons because it is used with `process.stdout.write`. I suggest a simpler more symmetrical pattern like:

```
    .on('help:header', () => {
      console.log('Custom Header\n');
    })
    .on('help:body', (instance) => {
      console.log('------------');
      process.stdout.write(instance.helpInformation()); // has trailing eol
      console.log('------------');
    })
    .on('help:footer', () => {
      console.log(`\nCommander - ${new Date().getFullYear()}`);
   })
```

Reply:
yeah, what you suggested was close to my first approach I believe, but I wanted the output to be neat. But indeed it is much better to read.